File: c-sharp-design-patterns/OOP Principles/Inheritance/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace c_sharp_design_patterns.OOP_Principles.Inheritance
{
    public class Vehicle
    {
        public string Brand { get; }
        public string Model { get; }
        public int Year { get; }

        public Vehicle(string brand, string model, int year)
        {
            Brand = brand;
            Model = model;
            Year = year;
        }


        public void Start()
        {
            Console.WriteLine("Vehicle is start...");
        }

        public void Stop()
        {
            Console.WriteLine("Vehicle is stop...");
        }
    }
}

File: OOP_Principles/Inheritance/Vehicle.java
package OOP_Principles.Inheritance;

public class Vehicle {

  protected String brand;
  protected String model;
  protected int year;

  public Vehicle(String brand, String model, int year) {
    this.brand = brand;
    this.model = model;
    this.year = year;
  }

  @Override
  public String toString() {
    return String.format("%d %s %s", this.year, this.brand, this.model);
  }

  public void Start() {
    System.out.println("Vehicle is start...");
  }

  public void Stop() {
    System.out.println("Vehicle is stop...");
  }
}
